astropy/coordinates/solar_system.py

```diff
@@ -228,10 +261,12 @@ def _get_body_barycentric_posvel(body, time, ephemeris=None,
             body_pv_bary = earth_pv_bary
 
         elif body == 'moon':
-            if get_velocity:
-                raise KeyError("the Moon's velocity cannot be calculated with "
-                               "the '{}' ephemeris.".format(ephemeris))
-            return calc_moon(time).cartesian
+            icrs_cart = calc_moon(time).cartesian
+            body_pos_bary = icrs_cart.without_differentials()
+            body_vel_bary = icrs_cart.differentials['s'].to_cartesian()
+            # TODO: the Meeus algorithm includes light travel time.
+            # That should be undo somewhere
```

Comment:
I note that, in fact, light travel time is omitted: 
```
**  2) This function is a full implementation of the algorithm
**     published by Meeus (see reference) except that the light-time
**     correction to the Moon's mean longitude has been omitted.
```
However, given the stated accuracy and the fact that the effect of light travel time to the geocenter amounts to only 0.7" in longitude, it is likely OK to omit the calculation of the light delayed position.

Reply:
Thanks for looking much more carefully! It actually is nicer - I think we might as well include the light travel time correction, making the case of the moon less special.